klpbuild/extractor.py

```diff
@@ -30,6 +31,9 @@ class Extractor(Config):
     def __init__(self, lp_name, lp_filter, apply_patches, app, avoid_ext, workers=4):
         super().__init__(lp_name, lp_filter)
 
+        self.sdir_lock = FileLock(Path(self.get_data_dir("x86_64"), "sdir.lock"))
```

Comment:
it should be using ARCH from klpbuild.utils. Otherwise it's good. Thanks!

Reply:
I hardcoded the "x86_64" because we only want to "lock" the src directory because that is the one being modified by the `--apply-patches` argument in the `extractor.py`. By checking all the code executed by using `--apply-patches,` the only src directory being patched is the "x86_64", which is currently also hard-coded. So, adding the same comment should be enough, IMO.

Reply:
I think so. The other solution maybe would be to download the source packages for all archs, now that we don't download the -extra packages (thus the amount of downloads is drastically reduced), but still I don't expect people to use the same data-dir as we do :)

Still thanks a lot for the patch!